modules/basic_auth/handler/basic_auth_logoff_handler.e

```diff
@@ -55,21 +55,62 @@ feature -- HTTP Methods
 			else
 				create {GENERIC_VIEW_CMS_RESPONSE} l_page.make (req, res, api)
 				unset_current_user (req)
-				l_page.set_status_code ({HTTP_STATUS_CODE}.found) -- Note: can not use {HTTP_STATUS_CODE}.unauthorized for redirection
-				if attached {WSF_STRING} req.query_parameter ("destination") as l_uri then
-					l_url := req.absolute_script_url (l_uri.url_encoded_value)
-				else
-					l_url := req.absolute_script_url ("")
-				end
+				l_page.set_status_code ({HTTP_STATUS_CODE}.unauthorized) -- Note: can not use {HTTP_STATUS_CODE}.unauthorized for redirection
+				l_url := req.absolute_script_url ("")
 				i := l_url.substring_index ("://", 1)
 				if i > 0 then
 						-- Note: this is a hack to have the logout effective on various browser
 						-- (firefox requires this).
 					l_url.replace_substring ("://_logout_basic_auth_@", i, i + 2)
 				end
-				l_page.set_redirection (l_url)
+				if
+					attached req.http_user_agent as l_user_agent and then
```

Comment:
I guess we would need here, to add to WSF the USER_AGENT class built for download module.